apps/worker/src/app/workflow/usecases/send-message/send-message-in-app.usecase.ts

```diff
@@ -201,12 +201,15 @@ export class SendMessageInApp extends SendMessageBase {
     const bridgeOutputs = command.bridgeData?.outputs as InAppOutput;
     const inAppMessage = inAppMessageFromBridgeOutputs(bridgeOutputs);
 
-    const channelData: Partial<Pick<MessageEntity, 'content' | 'subject' | 'avatar' | 'payload' | 'cta' | 'tags'>> = {
+    const channelData: Partial<
+      Pick<MessageEntity, 'content' | 'subject' | 'avatar' | 'payload' | 'cta' | 'tags' | 'data'>
+    > = {
       content: (this.storeContent() ? inAppMessage.content || content : null) as string,
       cta: bridgeOutputs ? inAppMessage.cta : step.template.cta,
       subject: inAppMessage.subject,
       avatar: inAppMessage.avatar,
       payload: messagePayload,
+      data: inAppMessage.data,
```

Comment:
This is the part of the code that I'm mostly not sure about. The `channelData` is the object that will be saved on the `message` document, however, it already contains the `payload` field, and I've added `data` field.

The thing that I'm not sure about is whether we want to feature flag this part like when v2 only saves the `data` but not `payload`, or should we keep both?

Reply:
Another question is whether we should allow for the `data` field for the other channels?